lib/Subscription.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Amp\Redis;
+
+use Amp\Deferred;
+use Amp\Promise;
+use Amp\Promisor;
+
+class Subscription implements Promise {
+    /** @var callable */
+    private $unsubscribeCallback;
+    /** @var Deferred */
```

Comment:
But isn't the typehint anyway implied in phpstorm redundant due to assign in ctor?